docs/reference/data-streams/change-mappings-and-settings.asciidoc

```diff
@@ -0,0 +1,520 @@
+====
+The following resolve index API request checks for any existing indices, index
+aliases, or data streams that start with `new_logs`. If not, the `new_logs*`
+wildcard pattern can be used to create a new data stream.
+
+[source,console]
+----
+GET /_resolve/index/new_logs*
+----
+// TEST[continued]
+
+The API returns the following response, indicating no existing targets match
+this pattern.
+
+[source,console-result]
+----
+{
+  "indices" : [ ],
+  "aliases" : [ ],
+  "data_streams" : [ ]
+}
+----
+====
+
+. Create or update a composable template. This template should contain the
+mappings and settings you'd like to apply to the new data stream's backing
+indices.
++
+This composable template must meet the
+<<create-a-data-stream-template,requirements for a data stream template>>. It
+should also contain your previously chosen name or wildcard pattern in the
+`index_patterns` property.
++
+TIP: If you are only adding or changing a few things, we recommend you create a
+new template by copying an existing one and modifying it as needed.
++
+.*Example*
+[%collapsible]
+====
+`logs_data_stream` is an existing composable template associated with the
+`logs` data stream.
+
+The following <<indices-templates,put composable template API>> request creates
+a new composable template, `new_logs_data_stream`. `new_logs_data_stream`
+uses the `logs_data_stream` template as its basis, with the following changes:
+
+* The `index_patterns` wildcard pattern matches any index or data stream
+  starting with `new_logs`.
+* The `@timestamp` field mapping uses the `date_nanos` field datatype rather
+  than the `date` datatype.
+* The template includes `sort.field` and `sort.order` index settings, which were
+  not in the original `logs_data_stream` template.
+
+[source,console]
+----
+PUT /_index_template/new_logs_data_stream
+{
+  "index_patterns": [ "new_logs*" ],
+  "data_stream": {
+    "timestamp_field": "@timestamp"
+  },
+  "template": {
+    "mappings": {
+      "properties": {
+        "@timestamp": {
+          "type": "date_nanos"                 <1>
+        }
+      }
+    },
+    "settings": {
+      "index.lifecycle.name": "logs_policy",
+      "sort.field" : [ "@timestamp"],          <2>
+      "sort.order" : [ "desc"]                 <3>
+    }
+  }
+}
+----
+// TEST[continued]
+
+<1>  Changes the `@timestamp` field mapping to the `date_nanos` field datatype.
+<2>  Adds the `sort.field` index setting.
+<3>  Adds the `sort.order` index setting.
+====
+
+. Use the <<indices-create-data-stream,create data stream API>> to manually
+create the new data stream. The name of the data stream must match the name or
+wildcard pattern defined in the new template's `index_patterns` property.
++
+IMPORTANT: If you do not want to mix new and old data, do not use an indexing
+request to create the data stream. Mixing old and new data is safe, but the
+backing index containing this data needs to be retained until you are ready to
+delete the new data.
```

Comment:
I don't understand this. Is this still related to reindexing into the new data stream? Is the new data the data that'll be sent directly to the new data stream (having not been ported over from the old data stream)? If that's the case where would we delete it from?

Reply:
Thanks for raising this @andreidan. I've revised this and the next relevant section.

To answer your question, I was envisioning a scenario in which a user creates the new data stream by indexing new documents to it. Because the user will later reindex data from an existing data stream into this new stream, it could result in one or more backing indices that contain mixed data.

This could interfere with data retention if the user deletes older backing indices. A user could prematurely delete an index they assume contains only old data but actually contain new data as well. This could result in unintended data loss.

The alternative is to keep such a backing indices until its newest data expires. This could increase storage costs and potentially violate their data retention policies.

Reply:
Thanks for the explanation and for revising this @jrodewig. It makes perfect sense now and I think it's great this sits in its own section now